File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.java
import java.util.*;

class Solution {
    public boolean solution(String[] phone_book) {
        boolean answer = true;
        HashMap<String, Integer> map = new HashMap<>();
        
        // 전화번호를 해시맵에 저장
        for (String phoneNumber : phone_book) {
            map.put(phoneNumber, 1);
        }
        
        // 각 전화번호에 대해 접두사 여부 확인
        for (String phoneNumber : phone_book) {
            for (int j = 1; j < phoneNumber.length(); j++) {
                if (map.containsKey(phoneNumber.substring(0, j))) {
                    return false;
                }
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.py
def solution(phone_book):
    hash_map = {}
    for pn in phone_book:
        hash_map[pn] = 1
        
    for pn in phone_book:
        jd = ""
        for n in pn:
            jd += n
            if jd in hash_map and jd != pn:
                return False
    return True